# Batch Inference And Artifacts

Load or train a model, score a batch, and write prediction files. This tests artifact writing and batch prediction reporting.

In [1]:
from pathlib import Path

exec(Path("samples/ml_tasks/batch_inference_artifacts.py").read_text(encoding="utf-8"))


model_source: loaded
model_artifact: artifacts/ml_tasks/baseline_model.json
prediction_csv: artifacts/ml_tasks/batch_predictions.csv
prediction_json: artifacts/ml_tasks/batch_predictions.json
predictions: 250
top_risk_ids:
- R00162: prob=0.896, label=1
- R00213: prob=0.881, label=1
- R00194: prob=0.863, label=1
- R00237: prob=0.857, label=1
- R00038: prob=0.849, label=1
- R00172: prob=0.838, label=1
- R00057: prob=0.835, label=1
- R00144: prob=0.820, label=1
- R00068: prob=0.796, label=1
- R00208: prob=0.794, label=1


## Finding: CLI artifact handling

Clean execution succeeded (`status: ok`, 1/1 code cells, 0.579s). The batch inference loaded `artifacts/ml_tasks/baseline_model.json`, wrote `artifacts/ml_tasks/batch_predictions.csv` and `artifacts/ml_tasks/batch_predictions.json`, and reported 250 predictions.

Artifact handling was visible enough to confirm this run because `show-cell --outputs full` preserved the artifact paths in stdout. For this workflow, the CLI would be more useful if `run-clean` or `show-cell` also surfaced a structured artifact summary with paths, existence, sizes, and a short preview, instead of requiring the user to parse stdout manually.
